### PR TITLE
kube-proxy: acquire the lock before flush list to avoid data race

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
kube-proxy: acquire the lock before flush list to avoid data race

**Which issue(s) this PR fixes**:

Ref #90540

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: pkg/proxy/ipvs/graceful_termination.go [hunk 1]
```diff
@@ -81,6 +81,9 @@ func (q *graceTerminateRSList) remove(rs *listItem) bool {
 }
 
 func (q *graceTerminateRSList) flushList(handler func(rsToDelete *listItem) (bool, error)) bool {
+	q.lock.Lock()
```
Comment (andrewsykim) [inline pkg/proxy/ipvs/graceful_termination.go:84]: I think this would result in a deadlock because the call `q.remove(rs)` also acquires this lock.

Comment (SataQiu) [reply pkg/proxy/ipvs/graceful_termination.go:84]: Thank you for pointing that out @andrewsykim 
Code has been updated.

**Review by cmluciano — COMMENTED** [review]
Can you please comment on the advantages of this approach versus the existing q.remove lock? If I'm understanding correctly, the race is potentially that newer things are added to the list before we run this graceful termination function. And looking at the code, I think the q.remove will provide that lock as is. Does that sound right @andrewsykim ?

**Comment by SataQiu** [comment]
Thanks @cmluciano 
/close

**Pull request closed without merging**